main.py

```diff
@@ -70,26 +70,30 @@ def set_random_seed(seed):
         mpu.model_parallel_cuda_manual_seed(seed)
 
 
-def get_losses(task, mse_weight, pearson_weight, gpu):
+def get_losses(task, mse_weight, pearson_weight, gpu, poisson_weight):
     """
     Return loss function. 
     Args:
         task : Whether the task is regression or classification or both.
         mse_weight : Mean squared error weight.
-        pearson_weight : Pearson weight.
+        pearson_weight : Pearson correlation loss weight.
+        poisson_weight : Poisson loss weight.
         gpu : Number of gpus.
     Return:
         loss_func : list of loss functions for each task.
     """
 
+    reg_loss_func = MultiLoss('poissonloss', poisson_weight, device=gpu) if poisson_weight > 0 
+                    else MultiLoss(['mse', 'pearsonloss'], [mse_weight, pearson_weight], device=gpu)
+    cla_loss_func = MultiLoss('bce', 1, device=gpu)
+
     if task == "regression":
-        loss_func = MultiLoss(['mse', 'pearsonloss'], [
-                              mse_weight, pearson_weight], device=gpu)
+        loss_func = reg_loss_func
     elif task == "classification":
-        loss_func = MultiLoss('bce', 1, device=gpu)
-    elif task == 'both':  # shouldn't reach here for now
-        loss_func = [MultiLoss(['mse', 'pearsonloss'], [mse_weight, pearson_weight], device=gpu),
-                     MultiLoss('bce', 1, device=gpu)]
+        loss_func = "cla_loss_func
```

Comment:
whoops, looks like there was a typo in my comment. there's an `"` which needs to be removed

Reply:
removed